src/components/Profile.js

```diff
@@ -60,43 +75,206 @@ class Profile extends Component {
     } catch (err) {
       console.error("error", err.message);
     }
+    try {
+      if (this.state.profileUser.likedRecipes.length === 0) {
+        this.setState({ loading: false });
+        return;
+      }
+      this.state.profileUser.likedRecipes.forEach(async likedRecipeId => {
+        const recipeDoc = await db
+          .collection("Recipes")
+          .doc(likedRecipeId)
+          .get();
+
+        this.setState({
+          likedRecipes: [...this.state.likedRecipes, recipeDoc.data()],
+          likedRecipeIds: [...this.state.likedRecipeIds, likedRecipeId],
+          loading: false
+        });
+      });
+    } catch (err) {
+      this.setState({ loading: false });
+      console.error("error", err.message);
+    }
   }
 
+  handleUpload = e => {
+    if (e.target.files[0] && e.target.files[0] !== this.state.image) {
+      const image = e.target.files[0];
+      this.setState(
+        () => ({ image }),
+        () => {
+          const { image } = this.state;
+          const uploadTask = storage.ref(`profileImages/${image.name}`).put(image);
+          uploadTask.on(
+            "state_changed",
+            snapshot => {
+              const progress = Math.round((snapshot.bytesTransferred / snapshot.totalBytes) * 100);
+              this.setState({ progress }, () => console.log(this.state.progress));
```

Comment:
```suggestion
              this.setState({ progress });
```
Der `console.log` kann noch raus c: